From 6cae372e14a1d3db47143ac01610e44a8d04d5f2 Mon Sep 17 00:00:00 2001
From: robcore <robpatershuk@gmail.com>
Date: Tue, 12 Apr 2016 18:05:18 -0600
Subject: [PATCH] From c49e43a8ebf2d478333ed16aeff3292bf437a62a Mon Sep 17
 00:00:00 2001 From: Slim80 <bestboss1980@gmail.com> Date: Wed, 23 Sep 2015
 16:02:07 +0200 Subject: [PATCH] readahead: make context readahead more
 conservative [Wu  Fengguang]

---
 mm/readahead.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mm/readahead.c b/mm/readahead.c
index 8129a68..6d61897 100755
--- a/mm/readahead.c
+++ b/mm/readahead.c
@@ -385,7 +385,7 @@ static int try_context_readahead(struct address_space *mapping,
 	 * no history pages:
 	 * it could be a random read
 	 */
-	if (!size)
+	if (size <= req_size)
 		return 0;

 	/*
@@ -396,8 +396,8 @@ static int try_context_readahead(struct address_space *mapping,
 		size *= 2;

 	ra->start = offset;
-	ra->size = get_init_ra_size(size + req_size, max);
-	ra->async_size = ra->size;
+	ra->size = min(size + req_size, max);
+	ra->async_size = 1;

 	return 1;
 }
---
 c49e43a8ebf2d478333ed16aeff3292bf437a62a.patch | 34 ++++++++++++++++++++++++++
 mm/readahead.c                                 |  6 ++---
 2 files changed, 37 insertions(+), 3 deletions(-)
 create mode 100644 c49e43a8ebf2d478333ed16aeff3292bf437a62a.patch

diff --git a/c49e43a8ebf2d478333ed16aeff3292bf437a62a.patch b/c49e43a8ebf2d478333ed16aeff3292bf437a62a.patch
new file mode 100644
index 0000000..d83390d
--- /dev/null
+++ b/c49e43a8ebf2d478333ed16aeff3292bf437a62a.patch
@@ -0,0 +1,34 @@
+From c49e43a8ebf2d478333ed16aeff3292bf437a62a Mon Sep 17 00:00:00 2001
+From: Slim80 <bestboss1980@gmail.com>
+Date: Wed, 23 Sep 2015 16:02:07 +0200
+Subject: [PATCH] readahead: make context readahead more conservative [Wu
+ Fengguang]
+
+---
+ mm/readahead.c | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/mm/readahead.c b/mm/readahead.c
+index 8129a68..6d61897 100755
+--- a/mm/readahead.c
++++ b/mm/readahead.c
+@@ -385,7 +385,7 @@ static int try_context_readahead(struct address_space *mapping,
+ 	 * no history pages:
+ 	 * it could be a random read
+ 	 */
+-	if (!size)
++	if (size <= req_size)
+ 		return 0;
+ 
+ 	/*
+@@ -396,8 +396,8 @@ static int try_context_readahead(struct address_space *mapping,
+ 		size *= 2;
+ 
+ 	ra->start = offset;
+-	ra->size = get_init_ra_size(size + req_size, max);
+-	ra->async_size = ra->size;
++	ra->size = min(size + req_size, max);
++	ra->async_size = 1;
+ 
+ 	return 1;
+ }
diff --git a/mm/readahead.c b/mm/readahead.c
index 6bb32a8..ad00ced 100644
--- a/mm/readahead.c
+++ b/mm/readahead.c
@@ -388,7 +388,7 @@ static int try_context_readahead(struct address_space *mapping,
 	 * no history pages:
 	 * it could be a random read
 	 */
-	if (!size)
+	if (size <= req_size)
 		return 0;
 
 	/*
@@ -399,8 +399,8 @@ static int try_context_readahead(struct address_space *mapping,
 		size *= 2;
 
 	ra->start = offset;
-	ra->size = get_init_ra_size(size + req_size, max);
-	ra->async_size = ra->size;
+	ra->size = min(size + req_size, max);
+	ra->async_size = 1;
 
 	return 1;
 }
